akka-camel/src/main/scala/akka/camel/Producer.scala

```diff
@@ -128,7 +128,7 @@ trait ProducerSupport extends Actor with CamelSupport {
     protected def produce(endpoint: Endpoint, processor: SendProcessor, msg: Any, pattern: ExchangePattern): Unit = {
       // Need copies of sender reference here since the callback could be done
       // later by another thread.
-      val producer = self
+      val producer = context.parent
```

Comment:
Perhaps it's enough to use the ordinary deployment configuration to define the dispatcher for such ProducerChild endpoints:
```
akka.actor.deployment {
  # use a dedicated dispatcher since this endpoint component might be blocking
  "/myproducer/*" {
    dispatcher = blocking-io-dispatcher
  }
}

blocking-io-dispatcher {
  type = Dispatcher
  executor = "thread-pool-executor"
  thread-pool-executor {
    fixed-pool-size = 32
  }
  throughput = 1
}
```